Add documentation for modbus.read_registers

ondras12345 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

  Before submitting your pull request, please verify that you have chosen the correct target branch,
  and that the PR preview looks fine and does not include unrelated changes.
-->
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
Added documentation for a new modbus service I implemented.



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/151769
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

ondras12345 committed:
Add documentation for modbus.read_registers

janiversen commented on source/_integrations/modbus.markdown:
```diff
@@ -1751,6 +1751,32 @@ data:
+
+| Action                | Description                 |
+| --------------------- | --------------------------- |
+| modbus.read_registers | Read register or registers  |
```
??? what is the difference between read register and read registers ? 

According to parameter explanation it always read registers with a count.

janiversen reviewed (changes requested):
A number of general problems:
- slave is replaced by device_id (services still accept unit).
- there are no difference between read_register and read_registers.

It is not documented what happens if the device returns an exception response.
It is not documented what happens if the device do not return anything (timeout).

ondras12345 committed:
Improve documentation for modbus.read_registers

ondras12345 committed:
Add note about use of modbus.read_registers

ondras12345 commented:
I have pushed a commit that addresses most of your concerns, except for the use of `slave`. Neither `device_id` nor `device_address` work with my code (or any of the other services in current Home Assistant releases). I will be happy to change that once https://github.com/home-assistant/core/pull/152565 is merged.
